include/deal.II/lac/arpack_solver.h

```diff
@@ -130,7 +139,40 @@ class ArpackSolver : public Subscriptor
 
   /**
    * Solve the generalized eigensprectrum problem $A x=\lambda B x$ by calling
-   * the <code>dneupd</code> and <code>dnaupd</code> functions of ARPACK.
+   * the <code>dneupd</code> and <code>dnaupd</code> functions of
+   * ARPACK.
+   *
+   * The function returns a vector of eigenvalues of length <i>n</i>
+   * and a vector of eigenvectors, where the latter should be twice
+   * the size of the eigenvalue vector. The first <i>n</i> vectors in
+   * <code>eigenvectors</code> will be the real parts of the
+   * eigenvectors, the second <i>n</i> the imaginary parts.
+   *
+   * @param A The operator for which we want to compute
+   * eigenvalues. Actually, this parameter is entirely unused.
+   *
+   * @param B The inner product of the underlying space, typically the
+   * mass matrix. For constrained problems, it can be a partial mass
+   * matrix, like for instance the velocity mass matrix of a Stokes
+   * problem. Only its function <code>vmult()</code> is used.
+   *
+   * @param inverse This is the possibly shifted inverse that is
+   * actually used instead of <code>A</code>. Only its function
+   * <code>vmult()</code> is used.
+   *
+   * @param eigenvalues is a vector of complex numbers in which the
+   * eigenvalues are returned.
+   *
+   * @param eigenvectors is a <b>real</b> vector of eigenvectors,
+   * containing alternatingly the real parts and the imaginary parts of the
+   * eigenvectors. Therefore, its length should be twice the number of
+   * eigenvalues. The vectors have to be initialized to match the
+   * matrices.
+   *
+   * @param n_eigenvalues The purpose of this parameter is not clear,
+   * but it is safe to set it to the size of <code>eigenvalues</code>
+   * or greater. Leave it at its default zero, which will be reset to the size
+   * of <code>eigenvalues</code> internally.
```

Comment:
I agree. If you don't want to break the interface,just say that the parameter is not used.